search.py

```diff
@@ -459,20 +469,13 @@ def run_search(dict_file, postings_file, queries_file, results_file):
     # 2. Process Queries
     with open(queries_file, "r") as q_file:
         with open(results_file, "w") as r_file:
-            for line in q_file:
-                # Each line is a query search
-                # Parse the initial query: filter punctuations, case-folding
-                line = line.rstrip("\n")
-                res = [] # contains all possible Postings for the result of a single line query
-                # try:
-                res = parse_query(line)
-                # except Exception as e:
-                    # print(repr(e), line)
-                    # res = [] # invalid search queries
-
-                # 3. Write the most relevant documents for the current query to storage
-                # in descending score order, then ascending doc_id
-                r_file.write(" ".join([str(r[1]) for r in res]) + "\n")
+            # TODO: Wrap these clause with Try catch block
+            lines = [line.rstrip("\n") for line in q_file.readlines()]
+            query = lines[0]
+            relevant_docids = [int(doc_id) for doc_id in lines[1:]]
+            res = []
+            res = parse_query(query, relevant_docids)
```

Comment:
Yup, got it, thanks!
This will make things much easier!

Reply:
can i merge?